core/services/bulletprooftxmanager/eth_confirmer_test.go

```diff
@@ -434,26 +459,71 @@ func TestEthConfirmer_FindEthTxsRequiringNewAttempt(t *testing.T) {
 	require.NoError(t, store.DB.Save(&attempt3_1).Error)
 
 	t.Run("returns the transaction if it is unconfirmed with an attempt that is older than gasBumpThreshold blocks", func(t *testing.T) {
-		etxs, err := bulletprooftxmanager.FindEthTxsRequiringNewAttempt(store.DB, currentHead, gasBumpThreshold)
+		etxs, err := bulletprooftxmanager.FindEthTxsRequiringNewAttempt(store.DB, address, currentHead, gasBumpThreshold, 10)
 		require.NoError(t, err)
 
 		require.Len(t, etxs, 2)
 		assert.Equal(t, etxWithoutAttempts.ID, etxs[0].ID)
 		assert.Equal(t, etx3.ID, etxs[1].ID)
 	})
 
+	t.Run("does not return more transactions the requested limit", func(t *testing.T) {
+		// Unconfirmed txes in DB are:
+		// (unnamed) (2)
```

Comment:
I see one confirmed tx on 367, one unnamed in-progress tx on 381, and one unnamed unconfirmed one on 392 but not two unnamed and unconfirmed transactions. Shouldn't that mean that a list of 4 should include `etx1`?

Reply:
No, it doesnt include etx1 because of lines 405 adn 408.

etx1 isnt eligible for gas bumping yet.

Reply:
Makes sense. Am I misreading the comment though? Why `unnamed (2)`? Seems like it should be 1, and then etx3 would be 5 and it would make sense why it is included when you request a limit of 5.

Reply:
the number represents the nonce